addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml

```diff
@@ -165,17 +159,68 @@ Resources:
                   - iam:AttachRolePolicy
                   - iam:DetachRolePolicy
                 Resource:
-                  - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${LaunchConstraintRolePrefix}'
-              - Effect: Allow
-                Action:
-                  - iam:CreatePolicy
-                  - iam:GetPolicy
-                  - iam:GetPolicyVersion
-                  - iam:ListPolicyVersions
-                  - iam:DeletePolicy
-                  - iam:DeletePolicyVersion
-                Resource:
-                  - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:policy/${LaunchConstraintPolicyPrefix}'
+                  - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${LaunchConstraintRolePrefix}LaunchConstraint'
+      PermissionsBoundary: !Ref CrossAccountEnvMgmtPermissionsBoundary
+
+  CrossAccountEnvMgmtPermissionsBoundary:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Permission boundary for cross account EnvMgmt role
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - s3:*
```

Comment:
Do we really need to grant _all_ permissions for these AWS actions? Could you mention the rationale - eg. template size management, risk related to user data etc.?

Reply:
This policy is only used as permission boundary, the actual permissions are much more restricted. I created the permission boundary broader to make it simpler and shorter.

Reply:
Fair enough. Since this doesn't have IAM related stuff, this is okay.